docs/expected_loss.ipynb

```diff
@@ -0,0 +1,203 @@
+    "\n",
+    "We are now interested in the expected loss $E(\\text{loss})$ a user suffers due to that. Loss is defined as losing access to their funds. In the federated setting that can be either due to theft or due to the adversary stopping the federation (which is easier). There exist countermeasures against adversaries merely stopping the federation (e.g. for ransom) but we will pessimistically assume that both failure modes mean loss of funds for the user (even if only for a limited time). In the centralized custodial case on the other hand the money gets stolen or not.\n",
+    "\n",
+    "We explore both federations of optimal sizes $(i \\cdot 3) + 1$ for as well as federations of suboptimal sizes. We see that adding members to a federation of already optimal size hurts resilience in our model since there are more potentially malicious parties but the threshold for how many malicious parties the system can handle does not change."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "prescribed-assumption",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def max_faulty_liveness(n):\n",
+    "    \"\"\"Returns maximum the number of malicious federation members for liveness\"\"\"\n",
+    "    return (n - 1) // 3\n",
+    "\n",
+    "def max_faulty_fund_safety(n):\n",
+    "    \"\"\"Returns maximum the number of malicious federation members for fund safety\"\"\"\n",
+    "    return n - max_faulty_liveness(n) - 1\n",
+    "\n",
+    "def p_mal_fail(n, mal, pmal):\n",
+    "    \"\"\"\n",
+    "    Returns the probability of mal of n members being malicious given the\n",
+    "    single-member-malicious probability pmal.\n",
+    "    \"\"\"\n",
+    "    return binomial(n, mal) * pmal**mal * (1 - pmal)**(n - mal)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "increased-survivor",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fedi_stall_prob(n, pmal):\n",
+    "    \"\"\"probability of the federation stalling\"\"\"\n",
+    "    return sum([p_mal_fail(n, mal, pmal) for mal in range(max_faulty_liveness(n) + 1, max_faulty_fund_safety(n) + 1)])\n",
+    "\n",
+    "def fedi_theft_prob(n, pmal):\n",
+    "    \"\"\"probability of the federation's funds getting stolen\"\"\"\n",
+    "    return sum([p_mal_fail(n, mal, pmal) for mal in range(max_faulty_fund_safety(n) + 1, n + 1)])\n",
+    "\n",
+    "def fedi_inaccessible_prob(n, pmal):\n",
+    "    \"\"\"probability of user funds in the federation being inaccessible\"\"\"\n",
+    "    return fedi_stall_prob(n, pmal) + fedi_theft_prob(n, pmal)\n",
```

Comment:
Is it correct to add probabilities here?

Reply:
I think so, the event `stall` has no overlap with event `theft` (`range` in python is end-exlusive).